### ISSUE TITLE
CORS error when fetching `jitsi.tld/.well-known/element/jitsi`

### ISSUE THREAD

**Issue by rwjack** [element-hq/element-web#24055]
### Steps to reproduce

1. Start video call
2. Get bunch of errors in console, starting with the inability to fetch `my-jitsi-host.tld/.well-known/element/jitsi`

```
Access to fetch at 'https://jitsi.domain.tld/.well-known/element/jitsi' from origin 'vector://vector' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

Which doesn't make sense because of the following:
```
root in xxx in ~ ❯ curl https://xxx/.well-known/element/jitsi -vvv
*   Trying xxx:443...
* Connected to xxx port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_128_GCM_SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=xxx
*  start date: Nov 30 11:37:50 2022 GMT
*  expire date: Feb 28 11:37:49 2023 GMT
*  subjectAltName: host "xxx" matched cert's "xxx"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x5586c1c032c0)
> GET /.well-known/element/jitsi HTTP/2
> Host: jitsi.xxx
> user-agent: curl/7.74.0
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200
< accept-ranges: bytes
< access-control-allow-origin: * ######################################## <------ It clearly is there
< content-type: application/json
< date: Mon, 19 Dec 2022 15:23:02 GMT
< etag: "63a06c23-1c"
< last-modified: Mon, 19 Dec 2022 13:50:27 GMT
< referrer-policy: same-origin
< server: Server
< strict-transport-security: max-age=31536000; includeSubDomains; preload
< x-content-type-options: nosniff
< x-forwarded-server: Server
< x-powered-by: Redstone
< x-xss-protection: 1; mode=block
< content-length: 28
<
{"auth": "openidtoken-jwt"}
* Connection #0 to host ... left intact
```

Relevant jitsi .env config:
```
MATRIX_UVS_URL=https://matrix.domain.tld # proxy forwards `/verify/user` and `/verify/user_in_room` to the matrix-UVS docker container
MATRIX_UVS_ISSUER=domain.tld # homeserver.yaml server_name
MATRIX_UVS_AUTH_TOKEN=syt_admin_user_token
MATRIX_UVS_SYNC_POWER_LEVELS=1
```

#### What did you expect?
For a jitsi conference to start
#### What happened instead?
Can't authenticate to jitsi using synapse users.

### Operating system

pop os

### Application version

Element version: 1.11.15 Olm version: 3.2.12

### How did you install the app?

https://element.io/download

### Homeserver

synapse docker container - {"server_version":"1.73.0","python_version":"3.9.15"}

### Will you send logs?

Yes

**Comment by rwjack** [comment]
Looking at: https://github.com/matrix-org/prosody-mod-auth-matrix-user-verification#widget-initialization

It seems that during the widget initialization, the json data isn't being sent to the Jitsi Meet external API in the correct format, hence the: 
```
jitsi_external_api.min.js:2 Failed to parse URL parameter value: jitsi.domain.tld SyntaxError: Unexpected token 'j', "jitsi.domain.tld" is not valid JSON
```
etc...

**Comment by gsouquet** [comment]
At first glance it looks like you need to configure your Jitsi instance to have proper CORS settings.
The rageshakes (logs) did not yield anything relevant here.

It is likely that the web server will return an error page in the HTML format for CORS errors, hence why you'd get the `is not valid JSON` error that you describe above

We would need more info, namely the jisti instance URL so that we can confirm that the CORS settings are correct

**Comment by rwjack** [comment]
Thanks @gsouquet , the jitsi URL is `jitsi.cthd.icu`.

Please let me know if you find anything relevant about the CORS settings that I could change.

**Comment by rwjack** [comment]
Also note that I can successfully authenticate when using Element iOS, so this seems to be an Element Desktop/Web issue.

**Comment by Beleggrodion** [comment]
It looks like i run into the same issue.  I tried it with synapse 1.74.0 and element-web 1.11.17
On the android element app it also hangs on the establishing the connection to the meeting server. Web and Desktop App wan'ts after definition of my meeting username a login for hosting the meeting or wait for the host of the meeting.

My used jitsi instance is based on the compose file from https://github.com/jitsi/docker-jitsi-meet with a nginx 

ENV variables:

```
MATRIX_UVS_URL=http://localhost:3000
MATRIX_UVS_ISSUER=matrix.domain.com
MATRIX_UVS_AUTH_TOKEN=sometoken
MATRIX_UVS_SYNC_POWER_LEVELS=1
```

My Nginx Settings are currently for testing really simple:

```
upstream jitsi-web {
        server 127.0.0.1:8880 fail_timeout=60s;
        keepalive 64;
}

server {
        server_name meet.domain.com;
        server_tokens off;

        access_log /var/log/nginx/jitsi_meet_access.log;
        error_log /var/log/nginx/jitsi_meet_error.log;

        location /.well-known/element/jitsi {
                add_header Content-Type application/json;
                add_header Access-Control-Allow-Origin * always;
                return 200 '{"auth": "openidtoken-jwt"}';
        }

        location / {
                 proxy_pass http://jitsi-web;
                 proxy_http_version 1.1;
                 proxy_set_header Upgrade $http_upgrade;
                 proxy_set_header Connection "upgrade";
                 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                 proxy_set_header X-Forwarded-Proto https;
                 proxy_set_header X-Real-IP $remote_addr;
        }

        location /xmpp-websocket {
                 proxy_pass http://jitsi-web;
                 proxy_http_version 1.1;
                 proxy_set_header Upgrade $http_upgrade;
                 proxy_set_header Connection "upgrade";
                 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                 proxy_set_header X-Forwarded-Proto https;
                 proxy_set_header X-Real-IP $remote_addr;

        }

        location /colibri-ws {
                 proxy_pass http://jitsi-web;
                 proxy_http_version 1.1;
                 proxy_set_header Upgrade $http_upgrade;
                 proxy_set_header Connection "upgrade";
                 proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                 proxy_set_header X-Forwarded-Proto https;
                 proxy_set_header X-Real-IP $remote_addr;
        }

        root /data/www/jitsi;

        listen 443 ssl http2;
	ssl_certificate /etc/letsencrypt/live/meet.domain.com/fullchain.pem; 
	ssl_certificate_key /etc/letsencrypt/live/meet.domain.com/privkey.pem; 
        include /etc/letsencrypt/options-ssl-nginx.conf; 
        ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; 
}
```

**Comment by rwjack** [comment]
@gsouquet @turt2live Hey guys, any update on this one?

**Comment by rwjack** [comment]
@gsouquet @turt2live Just letting you know, this is still broken.

**Comment by rwjack** [comment]
Still broken

**Comment by rwjack** [comment]
Still broken :c

Cannot use jitsi on my server until this is resolved. I also can't seem to find anyone else that has gotten Matrix UVS to work.

**Comment by rwjack** [comment]
@Beleggrodion did you manage to solve this?

**Comment by Beleggrodion** [comment]
Nope. The issue is still open here on my side but postponed due other stuff, so no progress since january.

**Comment by rwjack** [comment]
Closing this as we'll all be moving to EC once it's ready.
